### DISCUSSION TITLE
GetTriggerState() of Ischeduler return NORMAL for executing triggers also

### DISCUSSION THREAD

**Discussion by tejal2610** [quartznet/quartznet#1896] [Q&A]
In my project I am using multiple instances which are configured properly still I face concurrent job issue. To prevent that I am adding extra checks in my project. I am using IScheduler.GetTriggerState() method to get trigger states of triggers for one job key.
Some times this GetTriggerState() method misbehaves as it returns NORMAL state for all triggers of job key though some of them are actually executing.
I suspect that GetTriggerState() from [JobStoreSupport](https://github.com/quartznet/quartznet/blob/main/src/Quartz/Impl/AdoJobStore/JobStoreSupport.cs) class may not be matching with any AdoConstant and it is returning NORMAL.

Is there any problem with library? Please clarify things.

**Comment by lahma** [comment]
Might be a problem with your configuration, though you said it's "configured properly". Maybe you can create a failing test case so we can skip the possible guessing game?

**Reply by tejal2610** [reply]
For my project I have set these properties

"Quartz": {
    "quartz.scheduler.instanceId": "AUTO",
    "quartz.scheduler.instanceName": "**",
    "quartz.dataSource.myDs.provider": "**",
    "quartz.dataSource.myDs.connectionString": "**",
    "quartz.jobStore.type": "Quartz.Impl.AdoJobStore.JobStoreTX, Quartz",
    "quartz.jobStore.driverDelegateType": "**",
    "quartz.jobStore.dataSource": "**",
    "quartz.jobStore.useProperties": "**",
    "quartz.jobStore.clustered": "true",
    "quartz.serializer.type": "**"
  }

Each instance uses same property file. I think I have covered all the contraints.
InstanceId is unique.
clustered is set as true

Are there any any required properties?
